### ISSUE TITLE
Code indentation fix in rules

### ISSUE THREAD

**Issue by iromeo** [JetBrains-Research/snakecharm#26]
Code indentation fix in rules for new line after:
* rule name declaration
* rule parameter name

**Comment by winni2k** [comment]
yes please.

**Comment by winni2k** [comment]
:)

**Comment by iromeo** [comment]
Now next line is aligned relative to ':' and not line continuation is inserted. I think better to align rule parameter values relative to rule parameter name start pos + indent, but not clear how to fix this at the moment.

**Comment by winni2k** [comment]
Hey, I am not sure I understand everything, but this looks like it might be useful!

Just to re-iterate what I would expect to happen:
After I type a colon and hit enter, my cursor is sent to the next line and indented by four spaces compared to the previous line.

**Comment by iromeo** [comment]
I think not worse because we don't need '\' here and moving caret back to prev line to delete '\'  is more annoying.

**Comment by winni2k** [comment]
Hmm, currently when I start with this:

```
rule all:
    wildcard_constraints:<caret>
```

and hit enter, I get this:

```
rule all:
    wildcard_constraints:
    <caret>
```

I then have to hit <space> four times to get a sensible indent. Am I on an outdated version of the plugin?

**Comment by iromeo** [comment]
@winni2k Updated plugin is available in JetBrains Plugins Repository and could be installed using Preferences|Plugins

**Comment by winni2k** [comment]
Just a bit of feedback on our conversation above: I've started using this version of the plugin, and I love that the `code -> reformat code` option now works!

**Comment by winni2k** [comment]
I am still quite happy with this plugin, but I have a request for an improvement:

When I use `code -> reformat code` on the this code (note the indentation of two spaces before """cats"""):

```python
rule bla:
    shell:
      """cats"""
```

then the code is reformatted as:

```python
rule bla:
    shell:
         """cats"""
```

with an indentation of **five** spaces. 

I see this happening a lot. Is there any way to constrain the indents to be four spaces?

**Comment by iromeo** [comment]
@winni2k At the moment you cannot fix this. The issue is in aligning, `"""cats"""` is aligned according `shell` keyword end offset. I also don't like that section arguments get different indentation depending on it's section name length.

This issue is likely to be connected with #16. Snakemake language support reuses parts of `Python` plugin parser and code formater and at some points we cannot tune `python` code formatter to process `snakemake` syntax in the way which we want. We are going to work on these 2 issues during this summer.

**Comment by lpla** [comment]
Hi! Any progress on this?

**Comment by iromeo** [comment]
@lpla Yes, there is some progress, you can configure the formatter to keep constant indent independent on section keyword length. To do this:
* Uncheck checkbox `PyCharm | Preferences | Editor | Code Style | Python | Wrapping and Braces | Method call args | Align when multiline`

![image](https://user-images.githubusercontent.com/72933/70143038-a6178780-16ab-11ea-82ae-8ee9c854711a.png)

Then instead of
![image](https://user-images.githubusercontent.com/72933/70143090-cba49100-16ab-11ea-91ef-cfb004c82deb.png)

reformat code will behave like:
![image](https://user-images.githubusercontent.com/72933/70143148-eaa32300-16ab-11ea-9571-60f3223454eb.png)

The formatter is still not ideal, but to my mind, it works quite ok and this issue isn't critical at the moment.

**Comment by winni2k** [comment]
Thanks for the workaround! This has been driving me insane.

**Comment by iromeo** [comment]
Related API docs - https://www.jetbrains.org/intellij/sdk/docs/reference_guide/custom_language_support/code_formatting.html

**Comment by iromeo** [comment]
Snakemake specific codestyle settings paged added in coming 0.9.0
![image](https://user-images.githubusercontent.com/72933/102180920-b7988f80-3eba-11eb-9b65-3a6b177a8ec8.png)
